Source/options.cpp

```diff
@@ -1465,20 +1465,33 @@ void KeymapperOptions::KeyPressed(uint32_t key) const
 
 void KeymapperOptions::KeyReleased(uint32_t key) const
 {
+	if (key >= SDLK_a && key <= SDLK_z) {
+		key -= 'a' - 'A';
+	}
 	auto it = keyIDToAction.find(key);
 	if (it == keyIDToAction.end())
 		return; // Ignore unmapped keys.
 
 	const Action &action = it->second.get();
 
-	// Check that the action can be triggered and that the chat textbox is not
-	// open.
-	if (!action.actionReleased || (action.enable && !action.enable()) || ((talkflag || dropGoldFlag) && action.key != "Screenshot"))
+	// Check that the action can be triggered and that the chat or gold textbox is not
+	// open. If either of those textboxes are open, only return if the key can be used for entry into the box
+	if (!action.actionReleased || (action.enable && !action.enable()) || ((talkflag && IsTextEntryKey(key)) || (dropGoldFlag && IsNumberEntryKey(key))))
 		return;
 
 	action.actionReleased();
 }
 
+bool KeymapperOptions::IsTextEntryKey(SDL_Keycode vkey) const
+{
+	return ((vkey >= 0 && vkey < 128) || vkey == SDLK_LSHIFT || vkey == SDLK_RSHIFT || vkey == SDLK_KP_BACKSPACE);

```

Comment:
I'm a big ignorant to how this work so just asking. Does this take international symbols in to account or only American keyboards with a-Z?

Reply:
I'm not sure either. Could do the extended ASCII up to 255?

Reply:
Basically, it's all of these.
https://github.com/libsdl-org/SDL/blob/ac13ca9ab691e13e8eebe9684740ddcb0d716203/include/SDL_keycode.h#L52-L127

And also the delete key.
https://github.com/libsdl-org/SDL/blob/ac13ca9ab691e13e8eebe9684740ddcb0d716203/include/SDL_keycode.h#L150

Looking at the table, we might want to include caps lock, a bunch more of the `SDLK_KP_*` keys to support numeric keypads, and maybe also the `www` key (just in case?).

As for international symbols, most should be entered using the same keys located in the ASCII range so I think the 0-127 range is probably fine. We need to add `SDLK_RALT` since it seems that "Alt Gr" is used as a modifier to enter those symbols.

I'm not too sure about the more complex input systems, like those used for Japanese symbols. I do know that there are various keys for international keyboards to help enter those types of characters. SDL provides scancodes for these, but not keycodes. So we would likely receive those inputs as `SDLK_UNKNOWN`.

Reply:
Oh, and we should probably also include `SDLK_UP` and `SDLK_DOWN` as those can be used to cycle through message history.

Reply:
Looking at this more closely, it looks like you're only applying this function to `KeymapperOptions::KeyReleased()` because the print screen action only acts on key release and not key press. However, there is a similar function that filters (and acts on) key pressed events which should probably be kept consistent.

https://github.com/diasurgical/devilutionX/blob/5a12f7b0d543f7aed528c2d78b36e098f409d3de/Source/control.cpp#L1483-L1510

If we make any changes to better support international symbols or numeric keypads, we should apply them in both places.